static/js/publisher/release/promoteButton.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export default class PromoteButton extends Component {
+  componentDidMount() {
+    // use window instead of document, as React catches all events in document
+    window.addEventListener('click', this.closeAllDropdowns);
```

Comment:
This seems to be a pattern we'll need down the line. Juju-gui implemented a wrapper component that allowed the click-outside to be added easily to any component. We should think about doing something similar.

Reply:
I've seen some npm module for react that does more or less that, we can have a look, or see what juju implemented, or build our own component wrapper.